app/src/main/java/helper/CourseRequestHandler.java

```diff
@@ -149,10 +157,52 @@ public void onFailure(Call<List<CourseSection>> call, Throwable t) {
     }
 
 
+    private List<CourseSection> resolve(Object response) {
```

Comment:
This is a private method, and only used once in the project. So, the parameter can be `List<CourseSection>`, any specific reason for using `Object` ?

Reply:
The response.body() returns an object which is later typecast into List<CourseSection> inside the method.